test/cpu.cpp

```diff
@@ -1150,3 +1151,763 @@ TEST_CASE("TCD", "[cpu][instruction]") {
+	auto usingEmulatorMode = GENERATE(false, true);
+	
+	DYNAMIC_SECTION((usingEmulatorMode ? 8 : 16) << "-bit") {
+		testInstruction(Opcode::CLI, AddressingMode::Implied,
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = usingEmulatorMode ? 1 : 0;
+				cpu.setFlag(CPU::flags::i, false);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(cpu.getFlag(CPU::flags::i) == false);
+			}
+		);
+	}
+}
+
+TEST_CASE("CLD", "[cpu][instruction]") {
+	auto usingEmulatorMode = GENERATE(false, true);
+	
+	DYNAMIC_SECTION((usingEmulatorMode ? 8 : 16) << "-bit") {
+		testInstruction(Opcode::CLD, AddressingMode::Implied,
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = usingEmulatorMode ? 1 : 0;
+				cpu.setFlag(CPU::flags::d, false);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(cpu.getFlag(CPU::flags::d) == false);
+			}
+		);
+	}
+}
+
+TEST_CASE("SEC", "[cpu][instruction]") {
+	auto usingEmulatorMode = GENERATE(false, true);
+	
+	DYNAMIC_SECTION((usingEmulatorMode ? 8 : 16) << "-bit") {
+		testInstruction(Opcode::SEC, AddressingMode::Implied,
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = usingEmulatorMode ? 1 : 0;
+				cpu.setFlag(CPU::flags::c, true);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(cpu.getFlag(CPU::flags::c) == true);
+			}
+		);
+	}
+}
+
+TEST_CASE("SED", "[cpu][instruction]") {
+	auto usingEmulatorMode = GENERATE(false, true);
+	
+	DYNAMIC_SECTION((usingEmulatorMode ? 8 : 16) << "-bit") {
+		testInstruction(Opcode::SED, AddressingMode::Implied,
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = usingEmulatorMode ? 1 : 0;
+				cpu.setFlag(CPU::flags::d, true);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(cpu.getFlag(CPU::flags::d) == true);
+			}
+		);
+	}
+}
+
+TEST_CASE("SEI", "[cpu][instruction]") {
+	auto usingEmulatorMode = GENERATE(false, true);
+	
+	DYNAMIC_SECTION((usingEmulatorMode ? 8 : 16) << "-bit") {
+		testInstruction(Opcode::SEI, AddressingMode::Implied,
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = usingEmulatorMode ? 1 : 0;
+				cpu.setFlag(CPU::flags::i, true);
+			},
+			/*test=*/[&](CPU& cpu) {
+				REQUIRE(cpu.getFlag(CPU::flags::i) == true);
+			}
+		);
+	}
+}
+
+TEST_CASE("REP", "[cpu][instruction]") {
+	auto usingEmulatorMode = GENERATE(false, true);
+	auto valP = GENERATE_COPY(take(1, random<Address>(0, 0xff + 1)));
+	auto val = GENERATE_COPY(take(1, random<Address>(0, 0xff + 1)));
+	auto result = valP & ~val;
+	
+	DYNAMIC_SECTION((usingEmulatorMode ? 8 : 16) << "-bit") {
+		testInstruction(Opcode::REP, AddressingMode::Immediate,
+			/*addExpectedBusAccesses=*/noopAddBusAccesses,
+			/*setup=*/[&](CPU& cpu) {
+				cpu.e = usingEmulatorMode ? 1 : 0;
+				cpu.P = valP;
+				cpu.setFlag(CPU::flags::m, usingEmulatorMode);
+				cpu.setFlag(CPU::flags::x, usingEmulatorMode);
```

Comment:
Here, you're unconditionally setting the `m` and `x` flags to the value of `usingEmulatorMode`. However, if `usingEmulatorMode` is false, the CPU doesn't force those flags to be false, but you *are* forcing them to be false here (overwriting the value of those bits in `valP`).
```suggestion
				if (usingEmulatorMode) {
					cpu.setFlag(CPU::flags::m, true);
					cpu.setFlag(CPU::flags::x, true);
				}
```